README.md

```diff
@@ -735,16 +735,44 @@ This macro returns the sql required to build a date spine. The spine will includ
 ```
 
 #### deduplicate ([source](macros/sql/deduplicate.sql))
-This macro returns the sql required to remove duplicate rows from a model or source.
+This macro returns the sql required to remove duplicate rows from a model, source, or CTE.
+
+**Args:**
+ - `relation` (required): a [Relation](https://docs.getdbt.com/reference/dbt-classes#relation) (a `ref` or `source`) or string which identifies the model to deduplicate.
+ - `partition_by` (required): column names (or expressions) to use to identify a set/window of rows out of which to select one as the deduplicated row.
+ - `order_by` (required): column names (or expressions) that determine the priority order of which row should be chosen if there are duplicates (comma-separated string).
```

Comment:
Might I suggest an additional sentence here? "If there are multiple rows within the window specified by `partition_by` which share the same values for the columns specified in `order_by` then the result of deduplication may be nondeterministic over multiple runs of the model using the macro given that there is no way for the database to pick which row should 'win' the tie."

Here's an example of what I am talking about, it's for Snowflake, but it will be mathmatically equivalent for all other database platforms because it's not a specific database implementation thing, it's a math thing: https://community.snowflake.com/s/article/ROW-NUMBER-function-causing-Non-deterministic-behavior

Reply:
And I'm actually seeing this exact behavior in one of the current dbt projects that I am working on where someone used `ROW_NUMBER` to deduplicate but neglected to check if there could be "ties". So it's not just a rare or theoretical problem.

Reply:
Added a small note. Happy to call this resolved?

Reply:
This is great thank you so much :).